File: app/src/main/java/com/applify/securepass/data/VaultItem.java
package com.applify.securepass.data;

import java.util.UUID;

/**
 * Represents one saved password entry.
 * All fields are public for simplicity (or you can use getters/setters).
 */
public class VaultItem {
    public String id;           // Unique ID for each entry
    public String website;      // Website or app name
    public String username;     // Login / email
    public String password;     // The actual password (stored encrypted in file)
    public String notes;        // User notes, security questions, etc.
    public boolean isFavorite = false;
    public long createdAt;      // Timestamp when entry was created
    public long lastChanged;    // Timestamp when entry was last modified

    // Empty constructor required for Gson
    public VaultItem() { }

    // Convenience constructor
    public VaultItem(String website, String username, String password, String notes) {
        this.id = UUID.randomUUID().toString();
        this.website = website;
        this.username = username;
        this.password = password;
        this.notes = notes;
        this.createdAt = System.currentTimeMillis();
        this.lastChanged = this.createdAt;
    }
}

File: app/src/main/java/com/applify/securepass/data/VaultItem.kt
package com.applify.securepass.data

import java.util.UUID

/**
 * Represents one saved password entry.
 */
class VaultItem {
    @JvmField var id: String = ""
    @JvmField var website: String = ""
    @JvmField var username: String = ""
    @JvmField var password: String = ""
    @JvmField var notes: String = ""
    @JvmField var isFavorite: Boolean = false
    @JvmField var createdAt: Long = 0L
    @JvmField var lastChanged: Long = 0L

    // Empty constructor for Gson
    constructor()

    // Convenience constructor
    constructor(website: String, username: String, password: String, notes: String) {
        this.id = UUID.randomUUID().toString()
        this.website = website
        this.username = username
        this.password = password
        this.notes = notes
        this.createdAt = System.currentTimeMillis()
        this.lastChanged = this.createdAt
    }
}